content/docs/deploy/enterprise/configure-terraform.md

```diff
@@ -0,0 +1,108 @@
+---
+title: Configure with Terraform
+description: Learn how to manage your Pomerium Enterprise configuration using Terraform, including authentication setup, resource management, and deployment examples.
+keywords:
+  [
+    Pomerium Enterprise,
+    Terraform,
+    IaC,
+    infrastructure as code,
+    configuration management,
+    service accounts,
+    provider configuration,
+  ]
+---
+
+# Configure with Terraform
+
+Pomerium Enterprise can be configured and managed using Terraform through our official provider. This enables you to manage your Pomerium Enterprise resources as infrastructure as code, making it easier to version, review, and automate your configuration changes.
+
+## Provider Configuration
+
+To use the Pomerium Terraform provider, first configure it in your Terraform configuration:
+
+```hcl
+terraform {
+  required_providers {
+    pomerium = {
+      source  = "pomerium/pomerium"
+      version = "~> 0.0.7"
+    }
+  }
+}
+
+provider "pomerium" {
+  api_url = "https://console-api.your-domain.com"
+  # Choose one of the authentication methods below
+}
+```
+
+## Authentication Methods
+
+The provider supports two authentication methods:
+
+### 1. Service Account Token (Recommended)
+
+This method uses a [Pomerium Enterprise Service Account](/docs/capabilities/service-accounts) and provides fine-grained access control at the namespace level:
+
+```hcl
+provider "pomerium" {
+  api_url               = "https://console-api.your-domain.com"
+  service_account_token = var.pomerium_service_account_token
+}
+```
+
+### 2. Bootstrap Service Account
+
+This method requires enabling bootstrap service accounts in your Enterprise Console:
```

Comment:
I know the shared secret/bootstrap service account is not recommended, so just for clarity @wasaga, when you're back, the console API route always requires `allow_any_authenticated_user: true` if using the shared secret and boostrap service account? If so, can we reflect that in the docs?

Reply:
both methods are fine, its not like we're recommending one over the other. i removed the "Recommended" block and added policy descriptions.